src/Nethermind/Nethermind.Consensus/ExecutionRequests/ExecutionRequestProcessor.cs

```diff
@@ -0,0 +1,142 @@
+            {
+                flattenedResult.AddRange(byteArray);
+            }
+            else
+            {
+                throw new InvalidOperationException("Decoded ABI result contains non-byte array elements.");
+            }
+        }
+
+        // make sure the flattened result is of the correct size
+        if (flattenedResult.Count != depositRequestsBytesSize)
+        {
+            throw new InvalidOperationException($"Decoded ABI result has incorrect size. Expected {depositRequestsBytesSize} bytes, got {flattenedResult.Count} bytes.");
+        }
+
+        return new ExecutionRequest
+        {
+            RequestType = (byte)ExecutionRequestType.Deposit,
+            RequestData = flattenedResult.ToArray()
+        };
+    }
+
+
+    private IEnumerable<ExecutionRequest> ReadRequests(Block block, IWorldState state, IReleaseSpec spec, Address contractAddress)
+    {
+        bool IsWithdrawalRequests = contractAddress == spec.Eip7002ContractAddress;
+
+        int requestBytesSize = IsWithdrawalRequests ? withdrawalRequestsBytesSize : consolidationRequestsBytesSize;
+
+        if (!(IsWithdrawalRequests ? spec.WithdrawalRequestsEnabled : spec.ConsolidationRequestsEnabled))
+            yield break;
+
+        if (!state.AccountExists(contractAddress))
+            yield break;
+
+        CallOutputTracer tracer = new();
+
+        Transaction? transaction = new()
+        {
+            Value = UInt256.Zero,
+            Data = Array.Empty<byte>(),
+            To = contractAddress,
+            SenderAddress = Address.SystemUser,
+            GasLimit = GasLimit,
+            GasPrice = UInt256.Zero,
+        };
+        transaction.Hash = transaction.CalculateHash();
+
+        transactionProcessor.Execute(transaction, new BlockExecutionContext(block.Header), tracer);
+        var result = tracer.ReturnValue;
+        if (result == null || result.Length == 0)
+            yield break;
+
+        int requestCount = result.Length / requestBytesSize;
+
+        for (int i = 0; i < requestCount; i++)
+        {
+            int offset = i * requestBytesSize;
+            yield return new ExecutionRequest
+            {
+                RequestType = (byte)(IsWithdrawalRequests ? ExecutionRequestType.WithdrawalRequest : ExecutionRequestType.ConsolidationRequest),
+                RequestData = result.Slice(offset, requestBytesSize).ToArray()
+            };
+        }
+
+    }
+
+    public void ProcessExecutionRequests(Block block, IWorldState state, TxReceipt[] receipts, IReleaseSpec spec)
+    {
+        if (!spec.RequestsEnabled)
+            return;
+
+        List<ExecutionRequest> requests =
+        [
+            .. ProcessDeposits(receipts, spec),
+            .. ReadRequests(block, state, spec, spec.Eip7002ContractAddress),
+            .. ReadRequests(block, state, spec, spec.Eip7251ContractAddress),
+        ];
+        block.Header.RequestsHash = requests.ToArray().CalculateRoot();
```

Comment:
so we need all of this list and ExecutionRequest just to calculate root hash. It could be done in a lazy way, you don't need to create list and objects, just change CalculateRoot to lazy calculation